### ISSUE TITLE
Distributed Currency Exchange, Ricardian Contract

### ISSUE THREAD

**Issue by ABISprotocol** [OpenBazaar/OpenBazaar-Design#10]
See https://gist.github.com/drwasho/aa6ab79e92f2a876073e
My thinking is this could be developed into a simplified system for any crypto-crypto pair.
Please discuss

**Comment by gubatron** [comment]
That's be really cool.

Given the current state of the Bitcoin Blockchain scalability, supporting only about ~2,000 on-chain transactions every 10 minutes, or a mere 12,000 transactions worldwide per hour, OpenBazaar alone could put the network at risk in a few months from a stable release as Bitcoin rapidly gains popularity.

There's advanced ongoing work in the space of distributed exchanges, and even the Fiat part is covered, please take a look at the [BitSquare](https://github.com/gubatron/bitsquare) project, you can actually download and run the client today.

**Comment by ABISprotocol** [comment]
Yep, I read [your paper on scalability](https://medium.com/@gubatron/the-case-for-smart-bankers-a-federated-and-open-interbanking-blockchain-8add3119ad9c) with interest a while back, and I think you have some (really) good points.  In case you are curious, I've also read [this](https://medium.com/@drwasho/how-to-stop-a-bitcoin-hard-fork-or-how-to-accelerate-it-2197a08196ec#.c10jzktb6) and I don't want to advocate a doom and gloom approach, I do think a resolution of the situation is possible.   I've also been following closely the Bitsquare project, and I think it is an excellent model for decentralized currency exchange, though I'd like to see options within OB to make the Ricardian Contract for a crypto-crypto exchange easier to accomplish (in OB).

I do think the scalability issues will be worked out over the next several months to a year.  Some of the sentiment (votes by miners) on that [can be seen here right now](https://data.bitcoinity.org/bitcoin/block_size_votes/7d?c=block_size_votes&r=hour&t=bar).
With these graphs in mind as linked to above and the assumptions that the scalability issues will be addressed by a BIP (or more likely, at least a couple of BIPS over time), then I sense that it would be possible not only to see OB working now as a decentralized marketplace, as it is, but to also have an embedded option within OB so people can engage in utilizing buttons and drop-down menu selections to make the process of completing a Ricardian Contract easier.

This would involve:

(in order to create a currency Ricardian contract for a crypto-crypto exchange, which would require that one prepare a seed contract with the following data fields)

    Crypto-currency pair (e.g. Litecoin/Bitcoin; LTC/BTC)
    Exchange rate for the currency pair (e.g. 0.01637)
    Type (i.e buy, sell)
    Size (i.e. amount of the currency to buy/sell)
    Payment address

With that in mind, the prospective buyer needs to submit a bid offer by appending the following data to the seed contract:

    Buyer details
    Payment address

Drop-down fields for selection (e.g., LTC, BTC, etc), type, (buy, sell), and so forth, could be integrated; a specialized area, perhaps with a button, could be added ('Currency Exchange') if needed to orient persons to specific areas where the data is to be shared for that purpose.

You stated, "OpenBazaar alone could put the network at risk in a few months from a stable release as Bitcoin rapidly gains popularity."  I wish you were right, but I don't think that is the case.  I know that (approximately a year ago), when I was using the older, original (now deprecated) version of OB, there were 200 stores.  There's more now, and there's 1420 (or so) on Slack for OB.  With those numbers, to be honest, I don't think that OB will put the network at risk in a few months from a stable release.
